### PR TITLE
SI-8219 Source compatible name for implicit any2stringadd

### PR DESCRIPTION
**Pull request by retronym**
To support the established pattern for disabling it for
an compilation unit.

Review by @adriaanm

### CODE REVIEW

**Comment by adriaanm** [comment]
Please s/SI-8219/SI-8229/, then LGTM

File: src/library/scala/Predef.scala [hunk 1]
```diff
@@ -264,7 +264,8 @@ object Predef extends LowPriorityImplicits with DeprecatedPredef {
     @inline def formatted(fmtstr: String): String = fmtstr format self
   }
 
-  implicit final class StringAdd[A](private val self: A) extends AnyVal {
+  // SI-8229 retaining the pre 2.11 name for source compatibility in shadowing this implicit
+  implicit final class any2stringadd[A](private val self: A) extends AnyVal {
```
Comment (adriaanm) [inline src/library/scala/Predef.scala:268]: I was thinking about deprecating in the future. Maybe we could announce we intend to deprecate this in 2.12?
`@deprecated("Use String interpolation", "2.12.0")` would only be considered when running under -Xfuture. Next step: ignore deprecated implicits in implicit search, so people can opt-out of it easily.

Comment (adriaanm) [inline src/library/scala/Predef.scala:268]: A three-digit SI is calling... https://issues.scala-lang.org/browse/SI-194

Comment (retronym) [inline src/library/scala/Predef.scala:268]: Don't forget `Int.+(String)` and friends

Comment (gkossakowski) [inline src/library/scala/Predef.scala:268]: I think deprecating stringadd without a tool that rewrites users' code is not really good idea. Imagine our user that gets a sea of warnings and his only realistic choice is to disable them and recompile. He would think: why even bother warning if there's no practical way of addressing the problem?

Comment (adriaanm) [inline src/library/scala/Predef.scala:268]: Ah yes, good point!

Comment (adriaanm) [inline src/library/scala/Predef.scala:268]: That's why I would only enable the warning under -Xfuture and -deprecation

**Comment by adriaanm** [comment]
I'll look into the failures.

File: src/library/scala/Predef.scala (hunk 1 above)
Comment (soc) [inline src/library/scala/Predef.scala:268]: Agree. As soon as we have some way to see the warnings from the compiler, it is much easier to actually add IDE support for migrations to string interpolation.

Comment (retronym) [inline src/library/scala/Predef.scala:268]: Why? The IDE is free to traverse the tree to find use of `any2StringAdd` as a view today.

**Pull request closed without merging**